extensions/wikia/PowerUser/PowerUser.class.php

```diff
@@ -69,44 +86,141 @@ function __construct( \User $oUser ) {
 	}
 
 	/**
-	 * Adds a given PowerUser property to a user
+	 * Gets current PowerUser types for a user
+	 *
+	 * @return array
+	 */
+	public function getTypesForUser() {
+		$aUserTypes = [];
+		foreach ( self::$aPowerUserProperties as $sProperty ) {
+			if ( $this->oUser->isSpecificPowerUser( $sProperty ) ) {
+				$aUserTypes[] = $sProperty;
+			}
+		}
+		return $aUserTypes;
+	}
+
+	/**
+	 * Perform all actions to make a user a PowerUser
 	 *
 	 * @param string $sProperty One of the types in consts
 	 * @return bool
 	 */
 	public function addPowerUserProperty( $sProperty ) {
+		return ( $this->addPowerUserSetOption( $sProperty )
+			&& $this->addPowerUserAddGroup( $sProperty ) );
+	}
+
+	/**
+	 * Sets a specified PowerUser option to 1 in user_properties
+	 *
+	 * @param string $sProperty One of the types in consts
+	 * @return bool
+	 */
+	public function addPowerUserSetOption( $sProperty ) {
 		if ( in_array( $sProperty, self::$aPowerUserProperties ) ) {
 			$this->oUser->setOption( $sProperty, true );
 			$this->oUser->saveSettings();
-			$this->logSuccess( $sProperty, self::ACTION_ADD );
+			$this->logSuccess( $sProperty, self::ACTION_ADD_SET_OPTION );
 			return true;
 		} else {
-			$this->logError( $sProperty, self::ACTION_ADD );
+			$this->logError( $sProperty, self::ACTION_ADD_SET_OPTION );
 			return false;
 		}
 	}
 
 	/**
-	 * Removes a given PowerUser property from a user
+	 * Adds group rights to a user if the property's name
+	 * matches one in the aPowerUsersRightsMapping array
+	 *
+	 * @param string $sProperty One of the types in consts
+	 * @return bool
+	 */
+	public function addPowerUserAddGroup( $sProperty ) {
+		if ( in_array( $sProperty, self::$aPowerUsersRightsMapping )
+			&& !in_array( self::GROUP_NAME, $this->oUser->getGroups() ) )
+		{
```

Comment:
Closing bracket ```)``` should be before ```{``` (http://www.mediawiki.org/wiki/Manual:Coding_conventions#Line_continuation)